### ISSUE TITLE
CanvasItem.draw_rect drawing incorrectly with thick line width

### ISSUE THREAD

**Issue by briansemrau** [godotengine/godot#44312]
**Godot version:**

4.0 7195f6b973

**OS/device including version:**

RTX 2060, driver version 451.67, Vulkan

**Issue description:**

Using CanvasItem.draw_rect with filled=false and width=4.0f (in this case)
This is what it looks like:
![image](https://user-images.githubusercontent.com/6376721/101970293-96544c80-3bf7-11eb-9b7a-8fe6301dfea1.png)

**Steps to reproduce:**
In a node2d script:
```gdscript
func _draw():
	draw_rect(Rect2(0, 0, 20, 20), Color.orange, false, 4.0)
```

**Minimal reproduction project:**

[DrawRectMinRep.zip](https://github.com/godotengine/godot/files/5682362/DrawRectMinRep.zip)

**Comment by dalexeev** [comment]
A similar problem with lines thicker than 1 px. See PR #42183.

**Comment by dalexeev** [comment]
Fixed by #46677.

**Comment by akien-mga** [comment]
Thanks!
